File: javascript-web/samples/getCryptoCurrencies.js
var intrinioSDK = require('intrinio-sdk');
intrinioSDK.ApiClient.instance.authentications['ApiKeyAuth'].apiKey = "YOUR API KEY";
 
var cryptoAPI = new intrinioSDK.CryptoApi();
 
var opts = { 
  'exchange': "gemini" // String | Returns Crypto Currencies traded on the given Crypto Exchange.
};
 
cryptoAPI.getCryptoCurrencies(opts).then(function(data) {
  $('#apiResult').html(JSON.stringify(data, undefined, 2));
  
  var currencies = data.currencies;
  
  console.log(currencies.length + ' currencies found!');

  console.log('------------------------------------------------------------');

  currencies.forEach(function(currency) {
    console.log('Name:             ' + currency.name);
    console.log('Code:             ' + currency.code);
    console.log('Symbol:           ' + currency.symbol);
    console.log('First price date: ' + currency.first_price_date);
    console.log('Last price date:  ' + currency.last_price_date);
    console.log();
  });
}, function(error) {
  console.error(error);
});

File: ruby/samples/get_crypto_currencies.rb
# Load the gem
require 'intrinio-sdk'
 
# Setup authorization
Intrinio.configure do |config|
  config.api_key['api_key'] = 'YOUR API KEY'
end
 
crypto_api = Intrinio::CryptoApi.new
 
opts = { 
  exchange: "gemini" # String | Returns Crypto Currencies traded on the given Crypto Exchange.
}
 
begin
  result = crypto_api.get_crypto_currencies(opts)
  currencies = result.currencies
  
  if opts[:exchange]
    puts "#{currencies.size} curriences found for #{opts[:exchange]}!"
  else
    puts "#{currencies.size} currencies found!"
  end

  puts "------------------------------------------------------------"
  currencies.each do |currency|
    puts "Name:             #{currency.name}"
    puts "Code:             #{currency.code}"
    puts "Symbol:           #{currency.symbol}"
    puts "First price date: #{currency.first_price_date}"
    puts "Last price date:  #{currency.last_price_date}"
    puts ""
  end

rescue Intrinio::ApiError => e
  puts "Exception when calling CryptoApi->get_crypto_currencies: #{e}"
end
